### ISSUE TITLE
Site search

### ISSUE THREAD

**Issue by MitchellMarkGeorge** [MitchellMarkGeorge/TabButler#37]
As part of 1.0 (or further down the release calendar) it would we cool for users to search through different search engines and other sites (like stack overflow) directly from the extension.

**Comment by MitchellMarkGeorge** [comment]
Does this mean we should have two input fields (one for the search engine and one for the query)?

**Comment by andrei-g-git** [comment]
IMO having both a web search input and a tab/action search input on the same modal would over complicate the existing search framework which, to me, looks like is based on pulling in predefined ```Action``` items or a predetermined collection of tabs and listing them below the input. Having a 3rd modal with 1 or more search inputs sounds more straight forward without radically revamping the framework, if I understand it correctly.

**Comment by MitchellMarkGeorge** [comment]
@andrei-g-git agreed. with the sidebar shown in #36 I was thinking it would be a different SearchMode that can be triggered by the user (just like actions)

**Comment by andrei-g-git** [comment]
I sketched up something, tried to preserve the overall look of the extension:
![Web 1920 – 1 x](https://user-images.githubusercontent.com/77731047/179394745-0cc51cf4-4477-4be6-8003-3b76c472892f.png)

And I'm thinking it could have a feature to set the default borwser with a dropdown:



![Web 1920 – 2 x](https://user-images.githubusercontent.com/77731047/179394797-aa364d43-7727-4120-970d-9d7076966233.png)

...or a checkbox next to the search engine the user picks. The checkboxes and their label would appear on hover over the list item, except for the default engine where they are persistent. 

![Component 9 – 1](https://user-images.githubusercontent.com/77731047/179394851-c56db806-7ff8-4629-831c-22d6eea47207.png)

Let me know if you want a redo

**Comment by MitchellMarkGeorge** [comment]
@andrei-g-git Thanks so much for the mockups! I really like the checkbox idea. It fits into an idea I had for the regular tab search and will fit in nicely. Right now I would say to put off on working on this as I want to implements a "feature freeze" because of the beta launch coming up in 2 weeks. Once the freeze is over, I definitely believe this is one of the first features we should tackle alongside #36. Great work man! 🎖️

**Comment by andrei-g-git** [comment]
Alrighty off to q/a then *cracks fingers*

**Comment by MitchellMarkGeorge** [comment]
Moving this over to Relaunch milestone
